src/main/java/com/google/cloud/datafusion/plugin/successfactors/source/metadata/SuccessFactorsSchemaGenerator.java

```diff
@@ -0,0 +1,770 @@
+        .stream()
+        .map(this::buildSchemaField)
+        .collect(Collectors.toList());
+
+      String typeName = buildNestedTypeUniqueName(successFactorsColumnDetail.getName());
+
+      if (successFactorsColumnDetail.getType().equals(EdmTypeKind.ENTITY.name()) &&
+        (successFactorsColumnDetail.getMultiplicityOrdinal() != null &&
+          successFactorsColumnDetail.getMultiplicityOrdinal() > 0)) {
+        // adding 1 to * multiplicity record to ARRAY type
+        return Schema.Field.of(successFactorsColumnDetail.getName(),
+                               Schema.arrayOf(Schema.recordOf(typeName, outputSchema)));
+      }
+
+      // adding 0 to 1 multiplicity record to NULLABLE
+      return Schema.Field.of(successFactorsColumnDetail.getName(),
+                             Schema.nullableOf(Schema.recordOf(typeName, outputSchema)));
+    }
+
+    return Schema.Field.of(successFactorsColumnDetail.getName(), buildRequiredSchemaType(successFactorsColumnDetail));
+  }
+
+  /**
+   * Build and returns the appropriate schema type.
+   *
+   * @param successFactorsColumnDetail {@code SuccessFactorsColumnMetadata}
+   * @return {@code Schema}
+   */
+  private Schema buildRequiredSchemaType(SuccessFactorsColumnMetadata successFactorsColumnDetail) {
+    Schema schemaType = SCHEMA_TYPE_MAPPING.get(successFactorsColumnDetail.getType());
+
+    if (schemaType.getLogicalType() == Schema.LogicalType.DECIMAL
+      && successFactorsColumnDetail.getPrecision() != null
+      && successFactorsColumnDetail.getScale() != null) {
+
+      schemaType = Schema.decimalOf(successFactorsColumnDetail.getPrecision(), successFactorsColumnDetail.getScale());
+    }
+
+    // this check ensure that any DATE or TIME related fields are always set to NULLABLE Schema types.
+    // Reason: in SuccessFactors catalog service any DATE or TIME field which is mandatory can hold '00000000' in case
+    // of null and SuccessFactors service returns 'null' on data extraction for such fields so, to accordance this
+    // behaviour inside the plugin any DATE or TIME related Schema type are hardcoded to NULLABLE type.
+    if (schemaType.getLogicalType() == Schema.LogicalType.TIMESTAMP_MICROS ||
+      schemaType.getLogicalType() == Schema.LogicalType.TIME_MICROS) {
+
+      return Schema.nullableOf(schemaType);
+    }
+
+    return successFactorsColumnDetail.isNullable() ? Schema.nullableOf(schemaType) : schemaType;
+  }
+
+  /**
+   * Prepares a unique name for the provide name. This is required in case to avoid the same type name referencing
+   * issue at the runtime. Name format: <actualname>_<random name>
+   * e.g. Supplier_5810e28b_c38d_41fe_8dc1_e24150c515d9
+   *
+   * @param actualName nested property name
+   * @return unique name
+   */
+  private String buildNestedTypeUniqueName(String actualName) {
+
+    // schema name with '-', fails at runtime because '-' is not supported only character and '_' is supported
```

Comment:
nit: Change to 
`schema name with '-', fails at runtime because '-' is not supported. Only characters and '_' are supported`

Reply:
Changed. https://github.com/data-integrations/successfactors-plugins/commit/0ee84795bfe46404b5b855b83136b895898b170d